docs/core/atom.mdx

```diff
@@ -79,24 +82,24 @@ const Component = ({ value }) => {
 function atom<Value>(initialValue: Value): PrimitiveAtom<Value>
 
 // read-only atom
-function atom<Value>(read: (get: Getter) => Value | Promise<Value>): Atom<Value>
+function atom<Value>(read: (get: Getter) => Value): Atom<Value>
 
 // writable derived atom
-function atom<Value, Update>(
+function atom<Value, Args extends unknown[], Result>(
   read: (get: Getter) => Value | Promise<Value>,
-  write: (get: Getter, set: Setter, update: Update) => void | Promise<void>,
-): WritableAtom<Value, Update>
+  write: (get: Getter, set: Setter, ...args: Args) => Result,
+): WritableAtom<Value, Args, Result>
 
 // write-only derived atom
-function atom<Value, Update>(
+function atom<Value, Args extends unknown[], Result>(
   read: Value,
-  write: (get: Getter, set: Setter, update: Update) => void | Promise<void>,
-): WritableAtom<Value, Update>
+  write: (get: Getter, set: Setter, ...args: Args) => Result,
+): WritableAtom<Value, Args, Result>
 ```
 
 - `initialValue`: the initial value that the atom will return until its value is changed.
-- `read`: a function that's called on every re-render. The signature of `read` is `(get) => Value | Promise<Value>`, and `get` is a function that takes an atom config and returns its value stored in Provider as described below. Dependency is tracked, so if `get` is used for an atom at least once, the `read` will be reevaluated whenever the atom value is changed.
```

Comment:
`called on every re-render` is incorrect. atom's read will be only call once if it's a pure function like `atom(() => 1)`